packages/eslint-plugin/src/rules/consistent-type-exports.ts

```diff
@@ -214,6 +214,7 @@ function isSpecifierTypeBased(
   const symbol = checker.getSymbolAtLocation(node);
   const aliasedSymbol = checker.getAliasedSymbol(symbol!);
 
+  // eslint-disable-next-line @typescript-eslint/strict-enums
   if (!aliasedSymbol || aliasedSymbol.escapedName === 'unknown') {
```

Comment:
> One of the constituents of the __String type is itself a string (the first one).

It doesn't count as a string though because of the intersection. Strings cant have random object properties added to them - then they are no longer strings. Right?

Reply:
To my knowledge, the only case where a string would have a random object property added in the type system is this kind of type brand. Checking for that is outside the scope of the rule. It's also pretty rare for users to use type brands anyway.

Maybe another opt-in option? 😄.

Reply:
Ok, I'll try to make it work with intersection types.

However, I added this test:

```ts
enum InternalSymbolName {
  Foo = "foo",
  Bar = "bar",
}
type __String =
  | (string & { __escapedIdentifier: void; })
  | (void & { __escapedIdentifier: void; })
  | InternalSymbolName;
declare weirdString: __String;
if (weirdString === 'someArbitraryValue') {
}
```

And it passes, which surprised me.
Do you know how to write a test that triggers it?

Reply:
The following test passes now:

```ts
enum InternalSymbolName {
  Foo = "foo",
  Bar = "bar",
}
type __String =
  | (string & { __escapedIdentifier: void; })
  | (void & { __escapedIdentifier: void; })
  | InternalSymbolName;
declare const weirdString: __String;
if (weirdString === 'someArbitraryValue') {
}
```